src/repl/scala/tools/nsc/interpreter/MemberHandlers.scala

```diff
@@ -136,20 +136,28 @@ trait MemberHandlers {
           if (mods.isLazy) codegenln(false, "<lazy>")
           else any2stringOf(path, maxStringElements)
 
-        val nameString = string2code(prettyName) + (if (showObjIds) s"""" + f"@$${System.identityHashCode($path)}%8x" + """" else "")
+        val varOrValOrLzy =
+          if (mods.isMutable) "var"
+          else if (mods.isLazy) "lazy val"
+          else "val"
+
+        val nameString = {
+          string2code(prettyName) + (
+            if (showObjIds) s"""" + f"@$${System.identityHashCode($path)}%8x" + """"
+            else ""
+          )
+        }
+
         val typeString = string2code(req typeOf name)
-        s""" + "$nameString: $typeString = " + $resultString"""
+        s""" + "$varOrValOrLzy $nameString: $typeString = " + $resultString"""
       }
     }
   }
 
   class DefHandler(member: DefDef) extends MemberDefHandler(member) {
     override def definesValue = flattensToEmpty(member.vparamss) // true if 0-arity
-    override def resultExtractionCode(req: Request) = {
-      val nameString = string2code(name)
-      val typeString = string2code(req typeOf name)
-      if (mods.isPublic) s""" + "$nameString: $typeString\\n"""" else ""
-    }
+    override def resultExtractionCode(req: Request) =
+      if (mods.isPublic) codegenln(s"def ${req.defTypeOf(name)} = <method>") else ""
```

Comment:
Personally I don't like the `= <method>`; what does it give us over just leaving the RHS blank?

Reply:
I thought this would give some unity between values/variables which print a RHS, perhaps `lazy val` shouldn't have a RHS?

Reply:
> perhaps lazy val shouldn't have a RHS?

I'd be OK with that.

Reply:
fixed both of these